src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1494,60 +1494,128 @@ private static void ReportImplicitImplementationMatchDiagnostics(Symbol interfac
-                    implementedMember = implementedMethod.Construct(((MethodSymbol)implementingMember).TypeArgumentsWithAnnotations);
+                    var implementingEvent = (EventSymbol)implementingMember;
+                    var implementedEvent = (EventSymbol)interfaceMember;
+                    SourceMemberContainerTypeSymbol.CheckValidNullableEventOverride(compilation, implementedEvent, implementingEvent,
+                                                                                    diagnostics,
+                                                                                    (diagnostics, implementedEvent, implementingEvent, arg) =>
+                                                                                    {
+                                                                                        if (arg.isExplicit)
+                                                                                        {
+                                                                                            diagnostics.Add(ErrorCode.WRN_NullabilityMismatchInTypeOnExplicitImplementation,
+                                                                                                            implementingEvent.Locations[0], new FormattedSymbol(implementedEvent, SymbolDisplayFormat.MinimallyQualifiedFormat));
+                                                                                        }
+                                                                                        else
+                                                                                        {
+                                                                                            diagnostics.Add(ErrorCode.WRN_NullabilityMismatchInTypeOnImplicitImplementation,
+                                                                                                            GetImplicitImplementationDiagnosticLocation(implementedEvent, arg.implementingType, implementingEvent),
+                                                                                                            new FormattedSymbol(implementingEvent, SymbolDisplayFormat.MinimallyQualifiedFormat),
+                                                                                                            new FormattedSymbol(implementedEvent, SymbolDisplayFormat.MinimallyQualifiedFormat));
+                                                                                        }
+                                                                                    },
+                                                                                    (implementingType, isExplicit));
                 }
                 else
                 {
-                    implementedMember = interfaceMember;
-                }
+                    Action<DiagnosticBag, MethodSymbol, MethodSymbol, (TypeSymbol implementingType, bool isExplicit)> reportMismatchInReturnType =
```

Comment:
Consider making these local functions so we can make them static and easily verify that they are not capturing.

Reply:
+1 on that

---
In reply to: [296380417](https://github.com/dotnet/roslyn/pull/36663#discussion_r296380417) [](ancestors = 296380417)

Reply:
Delegates for lambdas are cached automatically.

---
In reply to: [296389943](https://github.com/dotnet/roslyn/pull/36663#discussion_r296389943) [](ancestors = 296389943,296380417)

Reply:
FWIW, for me it's less about caching and more about readability.

---
In reply to: [296390411](https://github.com/dotnet/roslyn/pull/36663#discussion_r296390411) [](ancestors = 296390411,296389943,296380417)